cpp/src/arrow/flight/test-util.cc

```diff
@@ -286,5 +287,72 @@ Status TestClientAuthHandler::GetToken(std::string* token) {
   return Status::OK();
 }
 
+Status GetTestResourceRoot(std::string* out) {
+  const char* c_root = std::getenv("ARROW_TEST_DATA");
+  if (!c_root) {
+    return Status::IOError("Test resources not found, set ARROW_TEST_DATA");
+  }
+  *out = std::string(c_root);
+  return Status::OK();
+}
+
+Status TestTlsCertificates(std::vector<CertKeyPair>* out) {
```

Comment:
Let's keep our existing naming conventions and call this `ExampleTlsCertificates`?